tstime/tstime.go

```diff
@@ -141,3 +142,71 @@ func Parse3339(s string) (time.Time, error) {
 func Parse3339B(b []byte) (time.Time, error) {
 	return parse3339m(mem.B(b))
 }
+
+func Append3339Nano(t time.Time, b []byte) []byte {
+	// Format: "2006-01-02T15:04:05.999999999Z07:00"
+	b = appendInt(b, int64(t.Year()), 4)
+	b = append(b, '-')
+	b = appendInt(b, int64(t.Month()), 2)
+	b = append(b, '-')
+	b = appendInt(b, int64(t.Day()), 2)
+	b = append(b, 'T')
+	b = appendInt(b, int64(t.Hour()), 2)
+	b = append(b, ':')
+	b = appendInt(b, int64(t.Minute()), 2)
+	b = append(b, ':')
+	b = appendInt(b, int64(t.Second()), 2)
+	if ns := t.Nanosecond(); ns != 0 {
+		b = append(b, '.')
+		b = appendInt(b, int64(ns), 9)
+		for b[len(b)-1] == '0' {
+			b = b[:len(b)-1]
+		}
+	}
+	_, zoneOffset := t.Zone()
+	if zoneOffset == 0 {
+		b = append(b, 'Z')
+		return b
+	}
+	if zoneOffset > 0 {
+		b = append(b, '+')
+	}
+	zoneOffset /= 60
+	b = appendInt(b, int64(zoneOffset/60), 2)
+	b = append(b, ':')
+	b = appendInt(b, int64(zoneOffset%60), 2)
+	return b
+}
+
+func appendInt(b []byte, n int64, width int) []byte {
```

Comment:
Aside: I've seen wrapper logic to `strconv.AppendUint` to ensure a fixed width many times that i wonder if the strconv package itself should support this.

Reply:
Yeah, that'd be nice.